src/NuGet.Clients/NuGet.CommandLine/Commands/RestoreCommand.cs

```diff
@@ -597,6 +597,25 @@ private PackageRestoreInputs DetermineRestoreInputs()
             return packageRestoreInputs;
         }
 
+        private bool IsSolutionOrProjectFile(string fileName)
+        {
+            if (!String.IsNullOrEmpty(fileName) && File.Exists(fileName))
+            {
+                string extension = Path.GetExtension(fileName);
```

Comment:
Path.GetExtension can return null if there is no extension
https://msdn.microsoft.com/en-us/library/system.io.path.getextension(v=vs.110).aspx